dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -4060,6 +4066,529 @@ public long pfcount(String... keys) {
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private boolean validHashtag(final String hashtag) {
+        return !Strings.isNullOrEmpty(hashtag) && hashtag.length() == 2;
+    }
+
+    private String ehashDataKey(String key) throws UnsupportedOperationException {
+        String hashtag = connPool.getConfiguration().getHashtag();
+        if (!validHashtag(hashtag)) {
+            throw new UnsupportedOperationException("hashtags not set");
```

Comment:
Why are we throwing an `UnsupportedOperationException` if the hashtag is not set and not another type of exception?

Reply:
`IllegalStateException` makes more sense instead.